### DISCUSSION TITLE
add : script which removes spaces between arrows'edges and shapes

### DISCUSSION THREAD

**Discussion by Yot67** [zsviczian/obsidian-excalidraw-plugin#522] [Ideas]
Kind of shape auto-join script to don't have spaces when shapes and arrows are still magneted.
![Obsidian_L6dHHaRABf](https://user-images.githubusercontent.com/14005233/158016037-28a54ace-c7d9-4d74-b72a-3c4cee902e43.png)
 TO
![Obsidian_oYhXAwl2v5](https://user-images.githubusercontent.com/14005233/158016049-387a0993-ba04-4761-ae3b-fff16b336ab3.png)

**Comment by zsviczian** [comment]
Hi @1-2-3, maybe this is something you could add to Elbow connectors?

**Comment by 1-2-3** [comment]
@zsviczian @Yot67 
I think I can add 2 settings to the `Elbow connectors` script: 1) `Change gap`, boolean type, when checked means the script will change the gap between the arrow and the box; 2) `Gap value`, integer, when 0 the arrow will be tight next to the box. What do you think of it?

**Comment by zsviczian** [comment]
I tried changing the gap value to zero manually, but the arrow did not get visibly closer to the box. I think it will require more than just changing the gap value, and will probably involve changing the points or the bounding box of the connector.

Here’s the example I tried:
![C1B3BF99-630C-4A42-9102-A2FE197237C4](https://user-images.githubusercontent.com/14358394/158074582-4a704eeb-f7df-41ef-a1e7-009fd6335f55.jpeg)

Before:
![A38354F4-B1DC-422E-9E7D-E01CAB8742DE](https://user-images.githubusercontent.com/14358394/158074580-d90a7128-fbbb-4886-ad4b-5538b20a3ff6.jpeg)

After:
![66F2767A-885A-4C23-93AB-D459DFB19C81](https://user-images.githubusercontent.com/14358394/158074577-1a18baf2-34d6-477d-9bf8-37b25792915c.jpeg)

The drawing did not visibly change.

**Comment by 1-2-3** [comment]
Hi @zsviczian , I tried changing the gap value to 0.1 manually, and need to drag the box with the mouse to trigger the refresh of the arrow position, the arrow can be close to the box. The gap value cannot be 0, otherwise the arrow will go into the center of the box.

**Comment by 1-2-3** [comment]
@Yot67 @zsviczian I added a Gap setting to the `Normalize selected arrows` script. Previously the script's gap value was 8 by default. Now the user can modify the gap value in the settings of the script. If the user wants the arrow to be next to the element, the gap value can be modified to 1. If the user wants to get elbow connectors, they can execute the `Normalize selected arrows` script first, and then execute the `Elbow connectors` script.
